### ISSUE TITLE
Implementation of mode switching

### ISSUE THREAD

**Issue by alexwkleung** [alexwkleung/Iris#19]
Iris was proposed to have mode switching (#6).

While this sounds good on paper, this can get complicated when the modes are isolated. Since the internals of the app aren't very flexible or dynamic, I would need to do a major refactor or rewrite. I would consider a rewrite or major refactor later on but at the moment I don't think it's worth it. 

Instead of isolated mode switching, it will be an editor switch instead. Basically the editor switching will behave exactly the same as the proposed mode switching except it won't be isolated. The folder and files will stay the same, except the editor. Users will need to take note of compatibility when editing their note in the basic or advanced editor.

Writing notes in the basic and advanced editors are interchangeable *only* if the note contains syntax up to the spec of CommonMark and/or doesn't contain custom syntax (HTML tags, GFM, diagram, math, etc). 

TL;DR:

The proposed mode switching will be replaced with an editor switch instead. The functionality will be the same except files aren't isolated between the editors.

**Comment by alexwkleung** [comment]
> Writing notes in the basic and advanced editors are interchangeable *only* if the note contains syntax up to the spec of CommonMark and/or doesn't contain custom syntax (HTML tags, GFM, diagram, math, etc). 

What I mean by this is that the basic editor won't be able to render custom syntax from the advanced editor. By default, the syntax that the basic editor cannot render will be shown as raw text of some sort.

To be clear here, the advanced editor "features" are powered through the reading mode (preview pane). So the advanced editor is just a way to enable writing custom syntax but the visuals are done through the reading mode toggle.

> If the reading mode powers the advanced features, would it make sense to add it as a dropdown option so those using the basic editor won't feel restricted in a way?

I agree with this. Adding the reading mode option in the kebab dropdown alongside basic and advanced would make the most sense. 

If I restricted usage of reading mode when a user is in the advanced editor, it wouldn't be good for those using the basic editor.

**Comment by alexwkleung** [comment]
In order for the basic editor to preserve the custom syntax from the advanced editor, I would need to modify the schema from `prosemirror-markdown` so it allows the nodes. I might directly copy the entire `prosemirror-markdown` package into the source code so I'll be able to customize it easily.

At the moment, any syntax that doesn't conform to the base schema will be completely removed from the file if it was edited in the advanced editor and then opened in the basic editor.

See https://discuss.prosemirror.net/t/unknown-node-types/655

**Comment by alexwkleung** [comment]
Basic mode will be replaced with a proposed custom WYSIWYG editor in a future release.
